### DISCUSSION TITLE
Installing RE on S5 while flashed with Hypfer

### DISCUSSION THREAD

**Discussion by sca075** [rand256/valetudo#513] [Q&A] [answered]
First of all, Hypfer was quite easy to setup on the vacuum, and many links in the docs are here somehow broken. 
I do understand that if I would reset the vacuum to factory default it will return to Xaiomi firmware waiting for setup.
In the procedure from Hypfer it is possible to use [valetudo-helper-miioota](https://github.com/Hypfer/valetudo-helper-miioota), to install the firmware pkg. I had as an idea to use the same procedure to flash Valetudo RE instead using the pkg of the last release.
Would it will work?

**Comment by pidator** [comment, marked as answer]
Afaik the valetudo-helper-miioota tool just helps you setting the right miio commands to install a pkg. So why not, just use the RE-pkg and it should work. It's the same procedure like pasting the miio update commands manually...

**Comment by sca075** [comment]
So here what to do to flash the vacuum with valetudo-helper-miioota.
1) Reset the vacuum to stock firmware, this depend from the [vacuum model](https://github.com/dgiese/dustcloud/wiki/Xiaomi-Vacuum-Robots-Factory-Reset)
2) using windows make sure to have PowerShell(386), create a folder where to copy valetudo-helper-miioota.exe and the packages with the MD5 files.
3) rename the vacuum_2034_valetudo_re_0_10_10.pkg to something like v11_002010.pkg
4) rename the vacuum_2034_valetudo_re_0_10_10.pkg.md5 to md5.txt
5) edit the md5.txt changing the name package name but do not touch the key.
7) verify that valetudo-helper-miioota runs.. and...
```
./valetudo-helper-miioota install-firmware v11_002010.pkg
Starting installer.
If you experience issues, make sure to disable your firewall and/or VPN.
Also, make sure that the robot is docked during the firmware update procedure.
If the install still fails, try turning the robot off and back on again and/or moving the laptop closer to it.

Robot discovery started...
Scan done.
Successfully discovered robot at 192.168.8.1
Reading firmware image..
Successfully read firmware image. Size: 78.62 MiB MD5Sum: fe820a713ec9efdfa3990b5d776e2cda

Listing for firmware download requests on http://192.168.8.10:34505/firmware
Response from robot: [ 'ok' ]
Received firmware download request from ::ffff:192.168.8.1..

Download seems to have finished.
The robot should now install the firmware. It will take 5-10 minutes.
Exiting..
```
easy way to flash it ;)
